canal/sync.go

```diff
@@ -245,9 +245,7 @@ func (c *Canal) updateReplicationDelay(ev *replication.BinlogEvent) {
 	atomic.StoreUint32(c.delay, newDelay)
 }
 
-func (c *Canal) handleRowsEvent(e *replication.BinlogEvent) error {
-	ev := e.Event.(*replication.RowsEvent)
-
+func (c *Canal) handleRowsEvent(e *replication.BinlogEvent, ev *replication.RowsEvent) error {
```

Comment:
we can change the parameter to (*EventHeader, *RowsEvent), that's not duplicated

Reply:
Yeah, you are right!

Reply:
As your suggestion, the parameters have been updated.